### ISSUE TITLE
dndmode not working properly

### ISSUE THREAD

**Issue by carlituxman** [krahabb/meross_lan#305]
Device info: 
mss310 6.0.0
Firmware: 6.3.21

When switch is on and dndmode off, after switch off the led stays on
When switch is off and dndmode off, after switch on the led stays off

**Comment by krahabb** [comment]
Well..I've right now tested mines (mss310 on old fw 2.something) and the light stays off when DNDMode is off whenever I switch the toggle.
So this could be an issue with the device fw itself or maybe some hw 'soft' failure.

Does the DND light turn on/off correctly when you operate the HA/meross_lan light entity ?

**Comment by carlituxman** [comment]
I have 3 device and all fail the same

I have the dndmode off and outlet is off, then switch outlet on but the led is off, then switch dndmode to on and turn on the led
When switch outlet off (with led on and dndmode on), then the led stay on, then switch dndmode off and then led turn off

Is like the dndmode need to be changed to refresh the led

**Comment by carlituxman** [comment]
Hi. Any news from this?

**Comment by krahabb** [comment]
I didn't have the time to reproduce the behavior you're experiencing so this is still 'in the queue'...
don't worry btw, I'll check it in the next few days ;)

**Comment by LDprg** [comment]
I hava a similar problem.

If the dnd is off and I turn the switch on the led stays off. If I then turn dnd on the led goes on. However If I now turn the switch off the light stays on and cant be turned off until I switch the switch on and change the dnd to off.

For me it looks like the led doesnt goes off when turning the switch off.

**Comment by carlituxman** [comment]
Any news about this??

**Comment by krahabb** [comment]
Not yet..still in the backlog ;)

**Comment by krahabb** [comment]
I did some tests and this is what happens on mines (mss310 fw 2.1.4)
- switch off
- dnd off (led is off)
- turn dnd on (led lights up in red)
- switch on (led lights up in red/blue)
- after a while the red goes away and just stays blue
- switch off (led off)
- switch on (led on blue)

It looks like whenever you turn on the dnd, the led goes in a 'lighted' state (mines go red) for a while and this is independent of the switch state. I think the led light has different meanings since it is also used to carry connection/pairing state so, when you turn it on (the dnd) the led not only shows the switch state but also some other info.

Could this explain the behavior you experience?

**Comment by carlituxman** [comment]
Not. Bassically the led not turn on. Only if change dnd turn on if switch on

**Comment by carlituxman** [comment]
> I did some tests and this is what happens on mines (mss310 fw 2.1.4)
> 
> * switch off
> * dnd off (led is off)
> * turn dnd on (led lights up in red)
> * switch on (led lights up in red/blue)
> * after a while the red goes away and just stays blue
> * switch off (led off)
> * switch on (led on blue)
> 
> It looks like whenever you turn on the dnd, the led goes in a 'lighted' state (mines go red) for a while and this is independent of the switch state. I think the led light has different meanings since it is also used to carry connection/pairing state so, when you turn it on (the dnd) the led not only shows the switch state but also some other info.
> 
> Could this explain the behavior you experience?

more or less yes

**Comment by helljol** [comment]
Issue still exist with mss710, I just switch from meross iot cloud integration to yours and the light does not switch on when the plug is set to on.

It works fine with the meross application too.

Anyway thanks for working on this integration, hope you can fix it in a future release

**Comment by Philou51** [comment]
Hello,
I just installed Meross-lan on HA and I have the same problem on 7 sockets mss210 7.0.0 (Firmware: 7.3.5)
The most annoying thing is that if my wife turns on a switch with her iPhone under Homekit, it turns on the LED correctly, and if I turn it off with HA, the LED stays on!
Thanks in advance for the fix

**Comment by andreaskoelsch** [comment]
I have a similar issue with the mss305. This is the behavior:

- Dnd switch off:
  - plug indicator light stays off regardless how I turn the plug on or off.
- Dnd switch on:
  - Turning on by pushing the hardware button or using the meross app turns on the light
  - Turning on using HA does not turn on the light
  - Turning off  by pushing the hardware button or using the meross app turns off the light
  - Turning off using HA does not turn off the light

My plug is on the most recent firmware 8.3.15, hardware version 8.0.0

To me, this looks like plug state and light state are independent and the app toggles both while this integration only toggles the plug state

**Comment by Sim209** [comment]
I have the same issue with my MSS310.

It looks like it works like this:
- If DND is off, the LED doesn't change.
- If DND is turned on, the LED copies the state of the plug (On or Off) and keeps that state even if the plug is turned off afterwards.

**Comment by wmutschl** [comment]
Hi, first of all: many many thanks for this. I had so many problems with using my plugs with matter and Apple, but noe they are all gone and it just works thanks to your work and home assistant. 

I do have one issue with the DND light, namely: if I turn off the plug in Home Assistant, the DND light stays green for about 5 minutes and then turns off (after the delay). Similarly, if I turn the plug on in Home Assistant there is a delay of 5 minutes before the DND light also turns on. 

The behavior is different when I use the Meross app or push manually the button: the DND light reflects correctly and directly the state of the plug. 

Btw: I have the newest firmware installed via the Meross app.

**Comment by krahabb** [comment]
I've just released a preview [version](https://github.com/krahabb/meross_lan/releases/tag/v5.6.0-alpha.1) with the fix

**Comment by wmutschl** [comment]
#576 fixes my issue with the delayed DND light! It all works now. Thank you very much!

**Comment by carlituxman** [comment]
Led light now works. But DND switch works invested, when dnd is ON leds works

**Comment by krahabb** [comment]
Yes @carlituxman, that's by 'design' in `meross_lan` since I've decided, back in the days, to represent the feature as a `light entity` representing the device 'lighting' which is nevertheless inverted in logic with respect to a 'common sense' DND feature.
